File: backend/gamification/src/main/java/ru/itmo/saferoad/gamification/domain/Achievement.java
package ru.itmo.saferoad.gamification.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.util.Map;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Data
@NoArgsConstructor
@Entity
@Table(name = "achievements")
public class Achievement {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false, unique = true)
  private String name;

  @Column(nullable = false)
  private String description;

  @Column(name = "icon_url", nullable = false)
  private String iconUrl;

  @JdbcTypeCode(SqlTypes.JSON)
  @Column(nullable = false, columnDefinition = "jsonb")
  private Map<String, Object> requirements;

  @Column(name = "reward_xp", nullable = false)
  private Integer rewardXp;

  @Column(name = "is_active", nullable = false)
  private Boolean isActive;
}

File: safe-road-mobile/lib/models/achievement.dart
class Achievement {
  final String title;
  final String description;
  final String iconUrl;
  final bool isUnlocked;
  final int rewardXp;

  Achievement({
    required this.title,
    required this.description,
    required this.iconUrl,
    required this.isUnlocked,
    required this.rewardXp,
  });

  factory Achievement.fromJson(Map<String, dynamic> json) => Achievement(
    title: json['title'],
    description: json['description'],
    iconUrl: json['iconUrl'],
    isUnlocked: json['isUnlocked'],
    rewardXp: json['rewardXp'],
  );
}
